internal/runbits/requirements/requirements.go

```diff
@@ -488,3 +500,71 @@ func packageCommitMessage(op bpModel.Operation, name, version string) string {
 	}
 	return locale.Tr(msgL10nKey, name, version)
 }
+
+// outputAdditionalRequirements computes and lists the additional dependencies being installed for the given package name.
+// This should only be called if a package or bundle is being added. Otherwise, the results may be nonsensical.
+func (r *RequirementOperation) outputAdditionalRequirements(parentCommitId, commitId strfmt.UUID, packageName string) (rerr error) {
+	pg := output.StartSpinner(r.Output, locale.T("progress_dependencies"), constants.TerminalAnimationInterval)
+	bp := model.NewBuildPlannerModel(r.Auth)
+
+	// Fetch old build plan with sources to compare against.
+	oldBuildPlan, err := bp.FetchBuildResult(parentCommitId, r.Project.Owner(), r.Project.Name())
```

Comment:
We should not be reaching out to the API for these buildplans, because those API's are known to be -slow-.

We have both the before and after buildplan stored locally, though there's probably an order of operations problem that'll need to be resolved. Point is though; this PR should not alter the performance of installing packages in any significant way.

Reply:
I've updated to use the cached build plan.

I am aware of the desire to avoid API calls and was cognizant of introducing performance issues. In my testing, I discovered that prior to these changes, when the runtime update gets to the "Resolving dependencies" stage, it's pretty slow (the buildplanner has to perform the solve, etc.). After these changes, this slowness is shifted prior to showing what dependencies are being added, but the subsequent runtime update stage is very fast (the buildplanner has already solved and is returning a cached result). Basically, we are shifting the solver slowness from at runtime update to showing additional dependencies.

This seems acceptable to me. The alternative would be to somehow modify the runtime to be able to receive information that a particular package was added, and if so, the runtime would to do this computation. That seems like a lot of unnecessary work in order to avoid a single API call for a cached buildplanner result.